### PR TITLE
Add Memory and CPUTime to Host interface

### PR DESCRIPTION
**Pull request by andrewkroh**
Implementations are added for Darwin, Linux, and Windows.

### CODE REVIEW

File: providers/darwin/host_darwin_amd64.go [hunk 1]
```diff
@@ -46,6 +46,87 @@ func (h *host) Info() types.HostInfo {
...
+	// Swap
+	swap, err := getSwapUsage()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get swap usage")
+	}
+
+	inactiveBytes := uint64(vmStat.Inactive_count) * pageSizeBytes
+	purgeableBytes := uint64(vmStat.Purgeable_count) * pageSizeBytes
+	mem.Metrics = map[string]uint64{
+		"active_bytes":         uint64(vmStat.Active_count) * pageSizeBytes,
```
Comment (ruflin) [inline providers/darwin/host_darwin_amd64.go]: My first thought was using `.bytes` for this but probably not the right place to do this here.

File: .ci/scripts/check_lint.go [hunk 2]
```diff
@@ -70,16 +70,14 @@ func main() {
...
 
 func filterIgnores(out []byte) ([]byte, error) {
-	var lines [][]byte
+	var buffer bytes.Buffer
 	s := bufio.NewScanner(bytes.NewReader(out))
 	for s.Scan() {
 		if !ignoreWarningsRe.Match(s.Bytes()) {
-			lines = append(lines, s.Bytes())
+			buffer.Write(s.Bytes())
+			buffer.WriteByte('\n')
```
Comment (ruflin) [inline .ci/scripts/check_lint.go]: That looks simpler then before 👍

File: providers/linux/boottime_linux.go [hunk 3]
```diff
@@ -22,23 +22,23 @@ import (
...
-	if !bootTime.IsZero() {
-		return bootTime, nil
+	if !bootTimeValue.IsZero() {
+		return bootTimeValue, nil
 	}
 
-	stat, err := procfs.NewStat()
+	stat, err := fs.NewStat()
 	if err != nil {
 		return time.Time{}, nil
```
Comment (ruflin) [inline providers/linux/boottime_linux.go]: Why not return the error err?

File: providers/darwin/host_darwin_amd64.go (hunk 1 above)
Comment (andrewkroh) [reply providers/darwin/host_darwin_amd64.go]: Yep, I was thinking that also. I was giving thought to creating a metric.Counter and metric.Gauge type that held the units. That would probably make it easier to consume and automatically add `.bytes` on the Beats side.

File: providers/linux/boottime_linux.go (hunk 3 above)
Comment (andrewkroh) [reply providers/linux/boottime_linux.go]: Looks like an omission on my part. Will fix.

File: providers/darwin/host_darwin_amd64.go (hunk 1 above)
Comment (ruflin) [reply providers/darwin/host_darwin_amd64.go]: Lets leave it as is for now as we can adjust it on the beats side manually for now.

**Pull request merged**